crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -629,17 +629,26 @@ pub fn derive_resource(input: TokenStream) -> TokenStream {
 /// On despawn, also despawn all related entities:
 /// ```ignore
 /// #[derive(Component)]
-/// #[relationship_target(relationship_target = Children, linked_spawn)]
+/// #[relationship_target(relationship = ChildOf, linked_spawn)]
 /// pub struct Children(Vec<Entity>);
 /// ```
 ///
+/// Allow relationships to point to their own entity:
+/// ```ignore
+/// #[derive(Component)]
+/// #[relationship(relationship_target = PeopleILike, allow_self)]
+/// pub struct LikedBy(pub Entity);
+/// ```
+/// **Warning:** When `allow_self` is enabled, be careful when using recursive traversal methods
+/// like `iter_ancestors` or `root_ancestor`, as they may loop infinitely if an entity points to itself.
```

Comment:
The whole PR uses the word "may" for this warning. Could you replace it with "will"?